### DISCUSSION TITLE
Assigning Wrong Permissions to Role

### DISCUSSION THREAD

**Discussion by ZERODayZServers** [spatie/laravel-permission#2307] [Q&A]
Hi all,

I've setup some basic roles and permissions and noticed that when assigning the permissions to the role it is assign the incorrect permission.

Please see an example below.

**Permission**
![image](https://user-images.githubusercontent.com/117211058/212933060-9f8e9da7-3b16-4c93-a5cb-1876a95f9cb2.png)

**Role**
![image](https://user-images.githubusercontent.com/117211058/212933396-98bf3514-d0e9-495e-9395-f5fa9578f0c1.png)

**Role Has Permission**
![image](https://user-images.githubusercontent.com/117211058/212933526-49510784-6247-4b55-af6b-089739e95954.png)

**Assigning the permission to role**
![image](https://user-images.githubusercontent.com/117211058/212933631-8346028d-2752-4272-9ae9-b106c67486e2.png)

The permission ID should be 1 but for some reason it's adding permission ID 5. Anyone ran into this issue before?

Thanks,
Nathan

**Comment by erikn69** [comment]
What is permission 5? I don't see it in the pictures
Also, laravel version? package version? almost no information
Are you using wildcards?

**Reply by erikn69** [reply]
> "spatie/laravel-permission" v5.5.5

Current is **5.8**
If you make manually changes to `Roles, Permissions` the cache will never reflect them, 

maybe it could be that, try `app()->make(\Spatie\Permission\PermissionRegistrar::class)->forgetCachedPermissions();` after `Role::updateOrCreate`

Also make a debug

**Comment by ZERODayZServers** [comment]
Permissiong ID 5 is:
![image](https://user-images.githubusercontent.com/117211058/212935639-cf1ba609-96df-41ed-956a-505ec2e72b2e.png)

I'm running:
"spatie/laravel-permission" v5.5.5
Laravel Framework 9.35.1

No wildcards are being used.
